org.civicrm.reqangular/src/common/components/notification-badge.component.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/components',
+  'common/services/api'
+], function (_, components) {
+  components.component('notificationBadge', {
+    bindings: {
+      filters: '<',
+      refreshCountEventName: '<'
```

Comment:
just a question about implementation: why isn't this a string instead? will it change?
```html
refresh-count-event-name="some-event"
```

Reply:
Yes, this is a string which will be passed on from the html. That is because in a page the notification badge can be present multiple times. Suppose once for Leave count, also for tasks or documents or something else. And each of them needs to be refreshed only when their respective count has changed(update, create, delete etc.).